src/providers/github/Github.ts

```diff
@@ -69,18 +77,32 @@ export class Github {
    * @param tag "v0.2.0", "release/patch"
    */
   async deleteTagIfExists(tag: string): Promise<void> {
+    if (!(await this.tagExists(tag))) return;
+
+    await this.octokit.request("DELETE /repos/{owner}/{repo}/git/refs/{ref}", {
+      owner: this.owner,
+      repo: this.repo,
+      ref: `tags/${tag}`,
+      headers: {
+        "X-GitHub-Api-Version": "2022-11-28"
+      }
+    });
+  }
+
+  private async tagExists(tag: string): Promise<boolean> {
     try {
-      await this.octokit.git.deleteRef({
+      await this.octokit.request("GET /repos/{owner}/{repo}/git/ref/{ref}", {
         owner: this.owner,
         repo: this.repo,
-        ref: `tags/${tag}`
+        ref: `tags/${tag}`,
+        headers: {
+          "X-GitHub-Api-Version": "2022-11-28"
+        }
       });
+      return true;
     } catch (e) {
-      // Ignore error if the reference does not exist, can be deleted latter
-      if (!e.message.includes("Reference does not exist")) {
-        e.message = `Error deleting tag ${tag}: ${e.message}`;
-        throw e;
-      }
+      if (e.status === 404 || e.status === "404") return false;
```

Comment:
same as above, check out oktokit types for better error handling

Reply:
Improved